### PR TITLE
add flyway-maven-plugin to Maven pom.xml

### PR DESCRIPTION
**Pull request by igor-zp**
add flyway-maven-plugin to Maven pom.xml 
and basic exception handling

### CODE REVIEW

File: src/main/java/com/pharmacy/optican/demo/exceptions/ApiException.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+@Getter
+public class ApiException {
+
+    private final String message;
+    private final Throwable throwable;
+    private final HttpStatus httpStatus;
+    private final ZonedDateTime timestamp;
+
+    public ApiException(String message, Throwable throwable, HttpStatus httpStatus, ZonedDateTime timestamp) {
```
Comment (EvgenKalashnikov) [inline src/main/java/com/pharmacy/optican/demo/exceptions/ApiException.java]: use Exeption istead throwable

File: src/main/java/com/pharmacy/optican/demo/exceptions/ApiRequestException.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.pharmacy.optican.demo.exceptions;
+
+public class ApiRequestException extends RuntimeException {
+
+    public ApiRequestException(ExceptionCode code) {
```
Comment (EvgenKalashnikov) [inline src/main/java/com/pharmacy/optican/demo/exceptions/ApiRequestException.java]: rename ExceptionCode to general exeption

File: src/main/java/com/pharmacy/optican/demo/exceptions/ExceptionCode.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import org.springframework.http.HttpStatus;
+
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+@ToString
+@Getter
+public enum ExceptionCode {
+
+    USER_NOT_FOUND("User is not present", HttpStatus.NOT_FOUND, ZonedDateTime.now(ZoneId.of("Z"))),
```
Comment (EvgenKalashnikov) [inline src/main/java/com/pharmacy/optican/demo/exceptions/ExceptionCode.java]: add field with exeption code

Comment (EvgenKalashnikov) [reply src/main/java/com/pharmacy/optican/demo/exceptions/ExceptionCode.java]: example GEEX001

File: src/main/java/com/pharmacy/optican/demo/exceptions/ApiException.java (hunk 1 above)
Comment (igor-zp) [reply src/main/java/com/pharmacy/optican/demo/exceptions/ApiException.java]: done

**Review by EvgenKalashnikov — APPROVED** [review]

**Pull request merged**